samples/net/http_server/src/main.c

```diff
@@ -159,12 +161,14 @@ static int led_update(uint8_t index, uint8_t state)
 		return -EBADMSG;
 	}
 
+#if defined(CONFIG_DK_LIBRARY)
 	ret = dk_set_led(index, led_states[index]);
 	if (ret) {
 		LOG_ERR("Failed to update LED %d state to %d", index, led_states[index]);
 		FATAL_ERROR();
 		return -EIO;
 	}
+#endif
```

Comment:
You could add `/* defined(CONFIG_DK_LIBRARY) */` here and other places

Reply:
Perhaps when plan to have `native_sim` board support in a sample, it might be easier to just avoid using LED/buttons altogether, so we avoid `#ifdefs` right and left, and in general having to maintain code that runs only on some of the boards that are supported by the sample.

It's nice to use LEDs for this sample when it's running on the kit, but it's not so nice to have `#ifdefs` in `main.c` files.
What if instead of a `/led/` endpoint this sample had a `/print/` endpoint? It could serve the purposes of the sample, be easier to verify in a test (instead of a LED turning on), and avoid all the LED-related code and `#ifdef` in `main.c`.

Reply:
Yeah, agree with the ifdefs. An alternative could be to have support for `native_sim` in the DK library and just let it print something like `LED 1: ON` or whatever

Reply:
That could work too